apps/src/lab2/views/Lab2ShareDialogWrapper.tsx

```diff
@@ -89,13 +91,21 @@ const Lab2ShareDialogWrapper: React.FunctionComponent<
 
   const onSubmitClick = () => {
     setDialogPanel('submit');
+    analyticsReporter.sendEvent(
+      EVENTS.SHARING_DIALOG_SUBMIT_TO_BE_FEATURED,
+      {
+        lab_type: projectType,
+        channel_id: channelId,
+      },
+      PLATFORMS.STATSIG
+    );
   };
 
-  if (!channelId || !projectType) {
+  if (!isDialogOpen || !channelId || !projectType) {
```

Comment:
Thanks @sanchitmalhotra126 for helping me investigate this error.
![Screenshot 2024-11-05 at 9 32 54 AM](https://github.com/user-attachments/assets/dd5a401b-a4b5-4809-a251-a73c47f50540)

Moving the check for `!isDialogOpen` here and returning early prevents the legacy `ShareDialog` from mounting momentarily when in a lab2 project.